beacon_node/network/src/sync/network_context.rs

```diff
@@ -441,6 +442,79 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
         active_request_count_by_peer
     }
 
+    /// Retries only the specified failed columns by requesting them again.
+    ///
+    /// Note: This function doesn't retry the whole batch, but retries specific requests within
+    /// the batch.
+    pub fn retry_columns_by_range(
+        &mut self,
+        request_id: Id,
```

Comment:
I tried never to pass just `Id` around to prevent mismatches and instead use the type `*RequestId` types. Since there are so many `Id` floating around a mismatch will cause this function to do nothing.